macros/zah.aegi-color-track.lua

```diff
@@ -59,14 +59,16 @@ local function showDialog(macro)
   end
 end
 
-local function getTimes(line)
-  local starterMS = line.start_time
-  local enderMS = line.end_time
-  local startframe = aegisub.frame_from_ms(starterMS)
-  local endframe = aegisub.frame_from_ms(enderMS)
-  local startMS = aegisub.ms_from_frame(startframe)
-  local endMS = aegisub.ms_from_frame(endframe)
+local function getFrames(line)
+  local startFrame = aegisub.frame_from_ms(line.start_time)
+  local endFrame = aegisub.frame_from_ms(line.end_time)
```

Comment:
For the endFrame, you cannot do that.

In the [Aegisub API](https://github.com/Aegisub/Aegisub/blob/master/src/auto4_lua.cpp#L166), you can see that it will return the START time

A little "hack" to convert start_time to end_time is to do that: ``local endFrame = aegisub.frame_from_ms(ms) - 1``

Also numOfFrames should be: ``local numOfFrames = endFrame - startFrame + 1``

Reply:
It's a question of semantics; if done as here, `endFrame` will be the first frame where the line _does not_ exist. This seems preferable to me, seeing as this is only used for the ffmpeg `-to` option, which stops reading the input at the specified time.

Though, I _just now_ noticed a problem in how `formatTimesFfmpeg` gets used (it expects timestamps, not frames). That one does definitely need fixing...

Reply:
> It's a question of semantics; if done as here, endFrame will be the first frame where the line does not exist.
If you see it like that, then my comment isn't good.
I said it since Aegisub include the endframe with the line timing
![image](https://user-images.githubusercontent.com/80980684/219192805-bad4ff03-f749-46fc-9994-cd3138b40033.png)